src/unix/newlib/mod.rs

```diff
@@ -242,6 +219,57 @@ s! {
     }
 }
 
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
+        pub type sigset_t = ::c_ulong;
+
+        s! {
+            pub struct stat {
```

Comment:
This is throwing some errors in the style check. What's the recommended way to have a different `stat` struct and `sigset_t` type for this target? The `s` macro doesn't allow `cfg_if` in its body, which is why we went with this approach.

Reply:
I think flattening it is a no-go because it doesn't match the toolchain. There might be padding/ABI issues (I assume).

Reply:
You need to move the Horizon-specific definitions to the `horizon/mod.rs` module. Generic definitions can go into a `generic` module used by non-horizon platforms.

Reply:
Updated to use a `generic` module.